pkgs/applications/graphics/mcomix3/default.nix

```
@@ -0,0 +1,86 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, python3
+, python3Packages
+, wrapGAppsHook
+, gobject-introspection
+, gtk3
+, gdk-pixbuf
+# Recommended Dependencies:
+, unrarSupport ? false
+, unrar
+, p7zip
+, lhasa
+, mupdf
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "mcomix3";
+  version = "20201123";
+
+  # fetch from github because no official release on pypi/github and no build system
+  src = fetchFromGitHub {
+    repo   = "${pname}";
+    owner  = "multiSnow";
+    rev = "cdcb27533dc7ee2ebf7b0a8ab5ba10e61c0b8ff8";
+    sha256 = "0q9xgl60ryf7qmy5vgzgfry4rvw5j9rb4d1ilxmpjmvm7dd3fm2k";
+  };
+
+  buildInputs = [ gobject-introspection p7zip lhasa mupdf ]
+    ++ lib.optional (unrarSupport) unrar;
+  nativeBuildInputs = [ wrapGAppsHook ];
+  propagatedBuildInputs = (with python3Packages; [ pillow pygobject3 pycairo ])
+    ++ [ gtk3 gdk-pixbuf ];
+
+  format = "other";
+  strictDeps = false;
+
+  preInstall = ''
+    libdir=$out/lib/${python3.libPrefix}/site-packages
+    mkdir -p $out/share/{icons/hicolor,man/man1,applications,metainfo,thumbnailers}
+    mkdir -p $out/bin $libdir
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ${python3.executable} installer.py --srcdir=mcomix --target=$libdir
+    mv $libdir/mcomix/mcomixstarter.py $out/bin/${pname}
+    mv $libdir/mcomix/comicthumb.py $out/bin/comicthump
+    mv $libdir/mcomix/mcomix/* $libdir/mcomix
+
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    rmdir $libdir/mcomix/mcomix
+    cp man/* $out/share/man/man1/
+    cp -r mime/icons/* $out/share/icons/hicolor/
+    cp mime/*.desktop $out/share/applications/
+    cp mime/*.appdata.xml $out/share/metainfo/
+    cp mime/*.thumbnailer $out/share/thumbnailers/
+    for folder in $out/share/icons/hicolor/*; do
+        mkdir $folder/{apps,mimetypes}
+        mv $folder/*.png $folder/mimetypes
+        cp $folder/mimetypes/application-x-cbt.png $folder/mimetypes/application-x-cbr.png
+        cp $folder/mimetypes/application-x-cbt.png $folder/mimetypes/application-x-cbz.png
+    done
+  '';
+
+  # real pytests seem to be broken upstream
+  checkPhase = ''
+    $out/bin/comicthump --help > /dev/null
+    $out/bin/${pname} --help > /dev/null
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Comic book reader and image viewer; python3 fork of mcomix";
+    longDescription = "User-friendly, customizable image viewer, specifically designed to handle comic books and manga supporting a variety of container formats (including CBR, CBZ, CB7, CBT, LHA and PDF)";
```

```suggestion
    longDescription = "User-friendly, customizable image viewer, specifically designed to handle comic books and manga supporting a variety of container formats (including CBR, CBZ, CB7, CBT, LHA and PDF)";
```

```suggestion
    longDescription = ''
--------------------------------------------------------------------------------     
      User-friendly, customizable image viewer, specifically designed to handle comic books 
      and manga supporting a variety of container formats (including CBR, CBZ, CB7, CBT, LHA and PDF)".
```

Use the two-quote notation **and** limit the lines to 80 columns:

---

```suggestion
    longDescription = ''
      User-friendly, customizable image viewer, specifically designed to handle
      comic books and manga supporting a variety of container formats
      (including CBR, CBZ, CB7, CBT, LHA and PDF)
    '';
```